extensions/wikia/AdEngine/js/AdLogicPageParams.js

```diff
@@ -241,8 +239,8 @@ define('ext.wikia.adEngine.adLogicPageParams', [
 		}
 
 		if (krux && !targeting.wikiDirectedAtChildren) {
-			params.u = krux.user;
-			params.ksgmnt = krux.segments && krux.segments.slice(0, maxNumberOfKruxSegments);
+			params.u = krux.getParams('user');
+			params.ksgmnt = krux.getParams('segments') && krux.getParams('segments').slice(0, krux.getDartSegmentsLimit());
```

Comment:
Or maybe, let's have a separate method for that: krux.getSegments (that would limit the number of segments) and krux.getUser